libs/server/Resp/Objects/SortedSetCommands.cs

```diff
@@ -878,5 +878,315 @@ private unsafe bool SortedSetDifferenceStore<TGarnetApi>(ref TGarnetApi storageA
 
             return true;
         }
+
+        /// <summary>
+        /// Computes an intersection operation between multiple sorted sets
+        /// and returns the result to the client.
+        /// </summary>
+        /// <param name="storageApi"></param>
+        /// <returns></returns>
+        private unsafe bool SortedSetIntersect<TGarnetApi>(ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            if (parseState.Count < 1)
+            {
+                return AbortWithWrongNumberOfArguments(nameof(RespCommand.ZINTER));
+            }
+
+            // Number of keys
+            if (!parseState.TryGetInt(0, out var nKeys))
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            if (parseState.Count < nKeys + 1)
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_SYNTAX_ERROR, ref dcurr, dend))
+                    SendAndReset();
+                return true;
+            }
+
+            var includeWithScores = false;
+            double[] weights = null;
+            var aggregateType = SortedSetAggregateType.Sum;
+            var currentArg = nKeys + 1;
+
+            // Read all the keys 
+            var keys = parseState.Parameters.Slice(1, nKeys);
+
+            // Parse optional arguments
+            while (currentArg < parseState.Count)
+            {
+                var arg = parseState.GetArgSliceByRef(currentArg).ReadOnlySpan;
+
+                if (arg.SequenceEqual(CmdStrings.WITHSCORES))
+                {
+                    includeWithScores = true;
+                    currentArg++;
+                }
+                else if (arg.SequenceEqual(CmdStrings.WEIGHTS))
+                {
+                    currentArg++;
+                    if (currentArg + nKeys > parseState.Count)
+                    {
+                        while (!RespWriteUtils.WriteError(CmdStrings.RESP_SYNTAX_ERROR, ref dcurr, dend))
+                            SendAndReset();
+                        return true;
+                    }
+
+                    weights = new double[nKeys];
+                    for (var i = 0; i < nKeys; i++)
+                    {
+                        if (!parseState.TryGetDouble(currentArg + i, out weights[i]))
+                        {
+                            while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_HASH_VALUE_IS_NOT_FLOAT, ref dcurr, dend))
```

Comment:
I will do this, but we don't have a consistent error message with Redis in most of the places. Is it mandated that all error messages should match Redis's error message? The error messages are not documented as well, will be hard to find each one for a command.

Reply:
We do try to adhere to the existing protocol, if possible. In this case it would be a more meaningful error as well. I agree that it's not very well documented, I usually have a Redis client open to see the all the possible error messages.